File: app/src/main/java/com/scatl/uestcbbs/module/board/adapter/ForumListLeftAdapter.java
package com.scatl.uestcbbs.module.board.adapter;

import android.widget.TextView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.ForumListBean;
import com.scatl.util.ColorUtil;

/**
 * author: sca_tl
 * description:
 * date: 2019/8/4 16:16
 */
public class ForumListLeftAdapter extends BaseQuickAdapter<ForumListBean.ListBean, BaseViewHolder> {

    private int selected = 0;

    public ForumListLeftAdapter(int layoutResId) {
        super(layoutResId);
    }

    public int getSelected() {
        return selected;
    }

    public void setSelected(int selected) {
        this.selected = selected;
        notifyDataSetChanged();
    }

    @Override
    protected void convert(BaseViewHolder helper, ForumListBean.ListBean item) {
        TextView name = helper.getView(R.id.forum_list_left_text);
        if (helper.getLayoutPosition() == selected) {
            name.setTextSize(18f);
            name.setTextColor(ColorUtil.getAttrColor(mContext, R.attr.colorPrimary));
        } else {
            name.setTextSize(15f);
            name.setTextColor(ColorUtil.getAttrColor(mContext, R.attr.colorOnSurfaceVariant));
        }
        helper.setText(R.id.forum_list_left_text, item.board_category_name);
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/board/adapter/ForumListLeftAdapter.kt
package com.scatl.uestcbbs.module.board.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.R
import com.scatl.uestcbbs.databinding.ItemForumListLeftBinding
import com.scatl.uestcbbs.entity.ForumListBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.util.ColorUtil.getAttrColor

/**
 * Created by sca_tl at 2023/6/9 15:20
 */
class ForumListLeftAdapter: PreloadAdapter<ForumListBean.ListBean, ItemForumListLeftBinding>() {

    var selected = 0
        set(value) {
            field = value
            notifyItemChanged(selected)
        }

    override fun getViewBinding(parent: ViewGroup): ItemForumListLeftBinding {
        return ItemForumListLeftBinding.inflate(LayoutInflater.from(parent.context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemForumListLeftBinding>, position: Int, item: ForumListBean.ListBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.text.text = item.board_category_name
        if (holder.layoutPosition == selected) {
            holder.binding.text.textSize = 18f
            holder.binding.text.setTextColor(getAttrColor(context, R.attr.colorPrimary))
        } else {
            holder.binding.text.textSize = 15f
            holder.binding.text.setTextColor(getAttrColor(context, R.attr.colorOnSurfaceVariant))
        }
    }
}